### PR TITLE
fix: do not require callback function for shellbar actions

### PR DESCRIPTION
**Pull request by mikerodonnell89**
#### Please provide a link to the associated issue.

#2084

### CODE REVIEW

**Comment by JKMarkowski** [comment]
Hello @mikerodonnell89, could you make your branch up to date with master? Or just cherry pick your own commits

**Comment by InnaAtanasova** [comment]
> Hello @mikerodonnell89, could you make this branch up to date with master? Or just cherry pick your own commits

+1

File: apps/docs/src/app/core/component-docs/info-label/info-label-docs.component.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+</fd-docs-section-title>
+<description> <code>fd-info-label fd-info-label--accent-color-1 fd-info-label--icon</code> modifier can be used to render a icon with text version of the Info Label. </description>
+<component-example>
+	<fd-info-label-text-icon-example></fd-info-label-text-icon-example>
+</component-example>
+<code-example [exampleFiles]="InfoLabelTextIconExample"></code-example>
+
+<separator></separator>
+
+<fd-docs-section-title [id]="'filled'" [componentName]="'infolabel'">
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/info-label/info-label-docs.component.html]: Already used (example above) `[id]` attribute value

File: apps/docs/src/app/core/component-docs/info-label/info-label-docs.component.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    selector: 'app-info-label',
+    templateUrl: './info-label-docs.component.html'
+})
+export class InfoLabelDocsComponent {
+
+    defaultInfoLabelHtmlType: ExampleFile[] = [
+        {
+            language: 'html',
+            code: InfoLableDefaultExample,
+            fileName: 'Info-label-default-example',
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/info-label/info-label-docs.component.ts]: I think it should be:
`fileName: 'Info-label-default-example',` => `fileName: 'info-label-default-example',`

File: libs/core/src/lib/info-label/info-label.component.html [hunk 3]
```diff
@@ -0,0 +1 @@
+<span class="fd-info-label" [ngClass]="buildComponentCssClass"><ng-content></ng-content></span>
```
Comment (salarenko) [inline libs/core/src/lib/info-label/info-label.component.html]: Is this template used anywhere?
info-label.component.ts:
```
@Component({
    // tslint:disable-next-line:component-selector
    selector: '[fd-info-label]',
    template: `<ng-content></ng-content>`,    <---------
    styleUrls: ['./info-label.component.scss'],
    encapsulation: ViewEncapsulation.None,
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class InfoLabelComponent implements OnInit, OnChanges, CssClassBuilder {
```

File: libs/core/src/lib/info-label/info-label.component.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    get labelType(): LabelType {
+        return this._labelType;
+    }
+
+    /** define the icon type */
+    _glyph: string; 
+
+    @Input()
```
Comment (salarenko) [inline libs/core/src/lib/info-label/info-label.component.ts]: Inputs need additional description for documentation purposes.

File: libs/core/src/lib/info-label/info-label.component.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef) {
+    }
+
+    ngOnInit(): void  {
+        this.buildComponentCssClass();
+    }
+
+    ngOnChanges(): void {
```
Comment (salarenko) [inline libs/core/src/lib/info-label/info-label.component.ts]: I think there is no point of using `ngOnChanges` only to trigger `this.buildComponentCssClass();` if every `@Input` does it by itself.

File: libs/core/src/lib/info-label/info-label.component.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            this._class
+        ].filter(x => x !== '').join(' ');
+    }
+
+    /** @hidden */
+    constructor(private _elementRef: ElementRef) {
+    }
+
+    ngOnInit(): void  {
+        this.buildComponentCssClass();
```
Comment (salarenko) [inline libs/core/src/lib/info-label/info-label.component.ts]: I think there is no point of using `ngOnInit` only to trigger `this.buildComponentCssClass();` if every `@Input` does it by itself.

File: apps/docs/src/app/core/component-docs/info-label/info-label-header/info-label-header.component.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<header>Info Label</header>
+<description>
+    <p>Info labels are used to indicate status. Colors, generally in combination with text, numeric and icon, are used to easily
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/info-label/info-label-header/info-label-header.component.html]: I fell like this description needs to be a little bit upgraded.

File: apps/docs/src/app/core/component-docs/info-label/info-label-docs.module.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import {InfoLabelHeaderComponent} from './info-label-header/info-label-header.component';
+import {InfoLabelModule} from '@fundamental-ngx/core';
+
+const routes: Routes = [
+    {
+        path: '',
+        component: InfoLabelHeaderComponent,
+        children: [
+            {path: '', component: InfoLabelDocsComponent},
+            {path: 'api', component: ApiComponent, data: {content: API_FILES.infolabel}}
```
Comment (salarenko) [inline apps/docs/src/app/core/component-docs/info-label/info-label-docs.module.ts]: Should be: `API_FILES.infolabel` => `API_FILES.infoLabel`

**Comment by salarenko** [comment]
My review comments are rather about InfoLabel component. Should these changes be in this PR?

**Pull request merged**